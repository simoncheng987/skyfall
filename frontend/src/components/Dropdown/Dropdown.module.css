.dropdownList {
  font-size: 32px;
  caret-color: transparent;
  text-align: center;
  border-radius: 15px;
  border-color: #1c1c1c;
  border-width: 3px;
  height: 55px;
  width: 100%;
}
