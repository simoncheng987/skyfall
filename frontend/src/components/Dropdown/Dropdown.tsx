import React from 'react';
import styles from './Dropdown.module.css';

interface DropdownProps {
  /**
   * The list of options to be displayed.
   */
  options: string[];

  /**
   * Action to be performed when value has been changed.
   */
  onChange?: React.ChangeEventHandler<HTMLSelectElement>;

  /**
   * Additional class names to style this component from the outside.
   */
  className?: string;
}

/**
 * A dropdown component that displays all the options provided.
 */
export default function Dropdown({ onChange, options, className }: DropdownProps) {
  return (
    <select onChange={onChange} className={`${styles.dropdownList} ${className}`}>
      {options.map((name) => (
        <option key={name} value={name}>
          {name}
        </option>
      ))}
    </select>
  );
}

Dropdown.defaultProps = {
  onChange: () => {},
  className: '',
};
