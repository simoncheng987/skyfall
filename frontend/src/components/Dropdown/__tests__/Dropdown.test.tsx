import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Dropdown from '../Dropdown';

describe('Dropdown', () => {
  const testOptions = ['option1', 'option2', 'option3'];
  const className = 'testClassName';

  it('renders all options', () => {
    render(<Dropdown options={testOptions} />);
    testOptions.forEach((str) => {
      expect(screen.getByText(str)).toBeInTheDocument();
    });
  });

  it('snapshot matches with className props', () => {
    const tree = renderer.create(<Dropdown options={testOptions} className={className} />).toTree();
    expect(tree).toMatchSnapshot();
  });
});
