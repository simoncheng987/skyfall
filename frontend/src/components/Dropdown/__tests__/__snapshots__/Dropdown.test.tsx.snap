// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown snapshot matches with className props 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "className": "testClassName",
    "onChange": [Function],
    "options": Array [
      "option1",
      "option2",
      "option3",
    ],
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <option
          value="option1"
        >
          option1
        </option>,
        <option
          value="option2"
        >
          option2
        </option>,
        <option
          value="option3"
        >
          option3
        </option>,
      ],
      "className": "dropdownList testClassName",
      "onChange": [Function],
    },
    "rendered": Array [
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "children": "option1",
          "value": "option1",
        },
        "rendered": Array [
          "option1",
        ],
        "type": "option",
      },
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "children": "option2",
          "value": "option2",
        },
        "rendered": Array [
          "option2",
        ],
        "type": "option",
      },
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "children": "option3",
          "value": "option3",
        },
        "rendered": Array [
          "option3",
        ],
        "type": "option",
      },
    ],
    "type": "select",
  },
  "type": [Function],
}
`;
